packages/solana/test/integration/staker.spec.ts

```diff
@@ -75,4 +79,25 @@ describe('SolanaStake - integration', () => {
     expect(newStakeAccount.state).to.equal('delegated')
     expect(newStakeAccount.amount).to.be.greaterThan(100000000) // we need to account for the rent exemption
   }).timeout(60000)
+  it('should unstake partial amounts', async () => {
+    const stakeAmount = 0.5
+    const stakeAccountAddress = await testStaker.createAndDelegateStake(stakeAmount.toString())
+    console.log(`Created stake account: ${stakeAccountAddress}`)
+
+    const stakeAccountsAfterCreation = await testStaker.getStakeAccounts(stakeAccountAddress)
+    expect(stakeAccountsAfterCreation.accounts.length).to.equal(1)
+    expect(stakeAccountsAfterCreation.accounts[0].state).to.equal('delegated')
+
+    const amountToUnstakeSOL = stakeAmount - 0.1
+    console.log(`Unstaking ${amountToUnstakeSOL} SOL`)
+    const status = await testStaker.undelegatePartialStake(amountToUnstakeSOL.toString())
+    expect(status.length).to.be.greaterThan(0)
+    status.forEach((s) => expect(s).to.equal('success'))
+    const stakeAccountsAfterSplit = await testStaker.getStakeAccounts(stakeAccountAddress)
```

Comment:
It would be great to assert various most common cases, say:
1. the amountToStake is tiny and there are 3 staked accounts, one with large sum, the other with mid and small. What we want to assert is that the small amount gets split
2. the amountToStake is mid and there are 3 staked accounts (small, mid, and large). The amountToStake == mid stake, so no need for split. Just simple unstake
3. the amountToStake is larger than two stake accounts combined, ensure the third account is split
etc ...

This is unittest from ton, where we check multiple cases:
https://github.com/ChorusOne/chorus-one-sdk/blob/main/packages/ton/test/pool-staker.spec.ts#L188